src/client/components/share-button/share-button.vue

```diff
@@ -0,0 +1,286 @@
+<template>
+  <div class="social-share">
+    <button class="social-share__button" @click="showSharingOptions"></button>
+
+    <div class="social-share__links">
+      <a :href="`https://twitter.com/home?status=${url}`"
+         :class="{ 'visible' : shareButtonsAreVisible }" 
+         class="social-share__share-link twitter"
+         target="_blank"
+         rel="noopener"
+         @click="handleClick('Twitter')">
+      </a>
+      
+      <a :href="`https://www.facebook.com/sharer/sharer.php?u=${url}`"
+         :class="{ 'visible' : shareButtonsAreVisible }" 
+         class="social-share__share-link facebook"
+         target="_blank"
+         rel="noopener"
+         @click="handleClick('Facebook')">
+      </a>
+
+      <a :href="`https://www.linkedin.com/shareArticle?mini=true&url=${url}&title=${title}&summary=${description}`"
+         :class="{ 'visible' : shareButtonsAreVisible }" 
+         class="social-share__share-link linkedin"
+         target="_blank"
+         rel="noopener"
+         @click="handleClick('Linkedin')">
+      </a>
+
+      <a :href="`whatsapp://send?text=${url}`"
+         data-action="share/whatsapp/share"
+         :class="{ 'visible' : shareButtonsAreVisible }" 
+         class="social-share__share-link whatsapp"
+         target="_blank"
+         rel="noopener"
+         @click="handleClick('WhatsApp')">
+      </a>
+
+      <a :href="`mailto:?&subject=${title}&body=${url}`"
+         :class="{ 'visible' : shareButtonsAreVisible }" 
+         class="social-share__share-link mail"
+         target="_blank"
+         rel="noopener"
+         @click="handleClick('Mail')">
+      </a>
+      
+      <button
+        v-if="copyToClipboardIsVisible"
+        :class="{ 'visible' : shareButtonsAreVisible }" 
+        class="social-share__share-link copy-to-clipboard"
+        @click="copyToClipboard">
+      </button>
+    </div>
+
+    <div :class="{ 'snackbar-visible' : snackbarIsVisible }" class="snackbar">{{ snackbarSuccessText }}</div>
```

Comment:
Should this be a separate component, so we can reuse it? If not, I guess the class name should be `social-share__snackbar` and we can just hardcode the success text, right?